libvirt/net/net_linux.go

```diff
@@ -599,3 +640,86 @@ func (c *Controller) ipReservationExists(network libvirt.ConnectNetworkShim, res
 
 	return false, nil
 }
+
+// releaseDHCPLease releases the DHCP lease associated to the reservation
+// if it exists.
+//
+// Implementation is inspired from:
+// https://github.com/imp/dnsmasq/blob/770bce967cfc9967273d0acfb3ea018fb7b17522/contrib/lease-tools/dhcp_release.c
+func (c *Controller) releaseDHCPLease(networkName, reservation string) error {
+	res := &libvirtxml.NetworkDHCPHost{}
+	if err := res.Unmarshal(reservation); err != nil {
+		return fmt.Errorf("failed to parse DHCP reservation: %w", err)
+	}
+
+	mac, err := stdnet.ParseMAC(res.MAC)
+	if err != nil {
+		return fmt.Errorf("failed to parse lease MAC address: %w", err)
+	}
+
+	network, err := c.netConn.LookupNetworkByName(networkName)
+	if err != nil {
+		return fmt.Errorf("failed to lookup network %q: %w", networkName, err)
+	}
+
+	bridge, err := network.GetBridgeName()
+	if err != nil {
+		return fmt.Errorf("failed to get bridge name for network %q: %w", networkName, err)
+	}
+
+	server, err := c.ipByInterfaceGetter(bridge)
+	if err != nil {
+		return fmt.Errorf("failed to get IP address for device %q: %w", bridge, err)
+	}
+
+	// Build a DHCP release packet to trigger releasing the lease.
+	pkt := layers.DHCPv4{
```

Comment:
Do we support IPv6 and use DHCPv6? If so do we need to check for that here?

Reply:
Correct, just ipv4 for now.